modules/client/src/main/java/org/apache/ignite/internal/client/TcpClientChannel.java

```diff
@@ -395,7 +419,7 @@ private <T> void complete(ClientRequestFuture<T> pendingReq, ClientMessageUnpack
     private void processNextMessage(ClientMessageUnpacker unpacker) throws IgniteException {
         if (protocolCtx == null) {
             // Process handshake.
-            complete(pendingReqs.remove(-1L), unpacker);
+            complete((ClientRequestFuture<?>) pendingReqs.remove(-1L).future(), unpacker);
```

Comment:
Let's make `TimeoutObject` generic with `T` being the type inside the `CompletableFuture`. This will allow to avoid these casts.